transformer_engine/common/include/transformer_engine/transformer_engine.h

```diff
@@ -41,6 +44,15 @@ struct NVTEShape {
   const size_t *data;
   /*! \brief Number of dimensions. */
   size_t ndim;
+
+  std::vector<size_t> to_vector() {
+    return std::vector<size_t>{data, data+ndim};
+  }
+
+  void from_vector(const std::vector<size_t> &shape) {
+    ndim = shape.size();
+    data = shape.data();
+  }
```

Comment:
```suggestion
```
I don't think these functions are used anywhere and they break the C API. Also, it seems easy to run into segfaults if the user calls `from_vector` using a temporary `std::vector`.

Reply:
Yeah, I think I was using them at an earlier stage and then forgot to strip the code out when they became unnecessary. Thanks for catching it!